test/index.js

```diff
@@ -19,96 +18,112 @@ const transform = (file, opts = {}) => (
   const {code} = await transform('./fixtures/stateless.js')
-  const out = await read('./fixtures/stateless.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('ignores whitespace around expression container', async t => {
   const {code} = await transform('./fixtures/whitespace.js')
-  const out = await read('./fixtures/whitespace.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('works with class', async t => {
   const {code} = await transform('./fixtures/class.js')
-  const out = await read('./fixtures/class.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('ignores when attribute is absent', async t => {
   const {code} = await transform('./fixtures/absent.js')
-  const out = await read('./fixtures/absent.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('works with global styles', async t => {
   const {code} = await transform('./fixtures/global.js')
-  const out = await read('./fixtures/global.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('generates source maps', async t => {
   const {code} = await transform('./fixtures/source-maps.js', {sourceMaps: true})
-  const out = await read('./fixtures/source-maps.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('mixed global and scoped', async t => {
   const {code} = await transform('./fixtures/mixed-global-scoped.js')
-  const out = await read('./fixtures/mixed-global-scoped.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('works with multiple jsx blocks', async t => {
   const {code} = await transform('./fixtures/multiple-jsx.js')
-  const out = await read('./fixtures/multiple-jsx.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('should not add the data-jsx attribute to components instances', async t => {
   const {code} = await transform('./fixtures/component-attribute.js')
-  const out = await read('./fixtures/component-attribute.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('works with expressions in template literals', async t => {
   const {code} = await transform('./fixtures/expressions.js')
-  const out = await read('./fixtures/expressions.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('should have different jsx ids', async t => {
   const {code} = await transform('./fixtures/different-jsx-ids.js')
-  const out = await read('./fixtures/different-jsx-ids.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('works with non styled-jsx styles', async t => {
   const {code} = await transform('./fixtures/non-styled-jsx-style.js')
-  const out = await read('./fixtures/non-styled-jsx-style.out.js')
-  t.is(code, out.trim())
+  t.snapshot(code)
 })
 
 test('throws when using `props` or constants ' +
   'defined in the closest scope', async t => {
-  [1, 2, 3, 4].forEach(i => {
-    t.throws(
-      transform(`./fixtures/invalid-expressions/${i}.js`),
-      SyntaxError
-    )
-  })
+  await t.throws(
```

Comment:
It has nothing to do with the string concatenation, it is because the test is `async`.
See: `[XO] Function was declared as async but doesn't use await (ava/no-async-fn-without-await)` 

I've tried some other solutions (using `await` in the `forEach`, using a `generator`, ...) but without success, this was the only solution that worked for me. Maybe you can steer me in the right direction if you have a solution?